how-it-works/1_protocol/02_consensus.subpage.md

```diff
@@ -5,14 +5,118 @@ shareImage: /img/how-it-works/consensus.600.jpg
+In case that the block from the rank-$0$ block maker cannot be fully notarized with a given time window, the rank-$1$ block maker will produce a block and also this block will be notarized by nodes, as outlined above.
+
+Nodes notary-sign blocks as long as they have not yet seen a fully-notarized block for this round.
+Once a node observes a fully notarized block for a round, it stops notarizing other blocks as then it is ensured by the P2P layer that eventually every node will receive this block and its notarization signature shares so that each of the nodes will be able to observe the full notarization of the block.
 
-The Internet Computer blockchain runs on a network of nodes in independent data centers located around the world. In each of its subnet blockchains, the participating nodes must process the same messages in the same order to ensure that they achieve the same state. To this end, the Internet Computer relies on a novel consensus protocol, as described in this presentation.
+If all goes well in a round, the block proposed by the first (the rank-$0$) block maker becomes fully notarized and no other block is even proposed by higher-rank block makers.
+This is the case when the consensus protocol consumes the least subnet resources.
+However, if things go wrong, e.g., network communication gets delayed or interrupted and the proposed block or notarization shares do not reach sufficiently many of the nodes in time, or the proposed block is invalid, further blocks may need to be proposed and notarized to finally get a fully-notarized block for this round.
+In this case, the resource consumption (networking bandwidth, CPU) and time to create the block for the round increases as multiple block proposals are created, broadcast, and notarized by nodes.
+The IC protocol ensures that the subnet performance degrades gracefully with increasing faults in the subnet communication or of nodes.
+
+[](/img/how-it-works/consensus_notarization.png)
+Notarization of increasing-rank block proposals in a round
+
+## Finalization
+
+Given the above-described scheme that allows up to all the nodes of a subnet to create and propose a block for a given round if things go wrong, more than one block can become fully notarized in a round.
+In this case it is not clear, which of the blocks should be the chosen one for this round.
+That means that we need an additional mechanism for *determining and observing finality*, i.e., which of the possibly multiple fully-notarized blocks of the round should be the one that is the chosen (finalized) block for the round which can be provided to the upper layers of the protocol stack to be executed.
+
+It is far from reality to assume a synchronous communication model in practice
```

Comment:
Wait, what? Why are talking about asynchrony here? Please just delete this sentence.